FFXIVClientStructs/FFXIV/Client/Game/UI/WeaponState.cs

```diff
@@ -1,21 +1,129 @@
+    /// <param name="isInstant">Whether the change should occur instantly.</param>
+    [MemberFunction("48 89 5C 24 ?? 48 89 6C 24 ?? 48 89 74 24 ?? 57 48 83 EC ?? 48 8B 3D ?? ?? ?? ?? 41 0F B6 E8 0F B6 DA")]
+    public partial void OnActorControlWeaponDrawn(ref byte stateFlag, byte newState, byte isInstant);
+
+    /// <summary>
+    /// Updates the state every frame.
+    /// This method decrements timers, resets them if necessary, and triggers auto-attack
+    /// disablement via the global auto-attack state.
+    /// </summary>
+    [MemberFunction("E8 ?? ?? ?? ?? 48 8D 8F ?? ?? ?? ?? E8 ?? ?? ?? ?? 48 8B 87 ?? ?? ?? ?? 48 8B 18")]
+    public partial void Tick();
+
+    /// <summary>
+    /// Sets the unsheathed state using the first method.
+    /// This method may trigger animations or effects as needed.
+    /// </summary>
+    /// <param name="newState">The new state value.</param>
+    /// <param name="sendPacket">Whether to send a network update.</param>
+    /// <param name="isInstant">Whether the change should occur instantly.</param>
+    [MemberFunction("E8 ?? ?? ?? ?? F6 44 24 ?? ?? 75")]
+    public partial bool SetUnsheathed(byte newState, byte sendPacket, byte isInstant);
+
+    /// <summary>
+    /// Sets the unsheathed state using the alternative variant (SetUnsheathed2).
+    /// This version may update state without sending network packets.
+    /// </summary>
+    [MemberFunction("E9 ?? ?? ?? ?? 48 8D 8E ?? ?? ?? ?? 40 84 FF")]
+    public partial bool SetUnsheathed2(byte newState);
 }
 
+/// <summary>
+/// Represents the AutoAttackState structure.
+/// Originally, this might have been embedded in WeaponState at offset 0x10,
+/// but current analysis shows auto-attack is managed globally.
+/// </summary>
 [GenerateInterop]
 [StructLayout(LayoutKind.Explicit, Size = 1)]
 public partial struct AutoAttackState {
-    [FieldOffset(0)] public bool IsAutoAttacking;
 
-    [MemberFunction("E8 ?? ?? ?? ?? EB 15 41 B0 01")]
-    public partial bool SetImpl(bool value, bool sendPacket, bool isInstant);
+    // Offset 0x00: A single byte that indicates if auto-attack is active.
+    [FieldOffset(0)]
+    public bool IsAutoAttacking;
+
+    /// <summary>
+    /// Gets the current auto-attack state.
+    /// </summary>
+    [MemberFunction("E8 ?? ?? ?? ?? 84 C0 74 ?? 38 1D")]
+    public partial bool Get();
+
+    /// <summary>
+    /// Handles actor control updates related to AutoAttack state.
+    /// Note: Internally accesses LocalPlayer at offset 0x9a0 in version 7.0 (previously 0x9b0).
+    /// </summary>
+    /// <param name="statePtr">Pointer to the byte representing the new state.</param>
+    /// <param name="newState">New state value.</param>
+    /// [MemberFunction("88 11 80 FA")] shorter but risky
+    [MemberFunction("88 11 80 FA 01 75 ?? 48 8B 0D ?? ?? ?? ?? 48 85 C9 74 ?? 38 15 ?? ?? ?? ?? 74 ?? 48 81 C1 ?? ?? ?? ??")]
+    public partial void OnActorControl(ref byte statePtr, byte newState);
```

Comment:
This function is defined wrong, it only has 1 arg for definition in C# where
`a2` is a bool of the new state